packages/flutter/lib/src/widgets/debug.dart

```diff
@@ -114,8 +114,17 @@ bool debugPrintGlobalKeyedWidgetLifecycle = false;
 ///  * [debugProfileLayoutsEnabled], which does something similar for layout,
 ///    and [debugPrintLayouts], its console equivalent.
 ///  * [debugProfilePaintsEnabled], which does something similar for painting.
+///  * [debugProfileBuildsEnabledUserWidgets], adds events for user generated
+///    [Widget] build.
 bool debugProfileBuildsEnabled = false;
 
+/// Adds [Timeline] events for every user created [Widget] built.
+///
+/// See also:
+///  * [debugProfileBuildsEnabled], functions similarly but shows events for
+///    every widget and has a higher overhead cost.
+bool debugProfileBuildsEnabledUserWidgets = true;
```

Comment:
this should be added to `debugAssertAllWidgetVarsUnset`

Reply:
Can we not default this to true?

Alternatively, if we do default this true, please provide some numbers on how this impacts internal benchmarks we are tracking. The idea is we should know exactly how much this regresses/adds noise to those benchmarks.

Reply:
> this should be added to debugAssertAllWidgetVarsUnset

Done

Reply:
> Can we not default this to true?\
\
Alternatively, if we do default this true, please provide some numbers on how this impacts internal benchmarks we are tracking. The idea is we should know exactly how much this regresses/adds noise to those benchmarks.

I added benchmarks.  The discussion about defaulting this to false is below.  We'll see if we can get it a bit faster.  There is Jacob's idea but that is a pretty big change too.